openfl-tutorials/interactive_api/jax_linear_regression/workspace/custom_adapter.py

```diff
@@ -0,0 +1,21 @@
+# Copyright (C) 2020-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+"""Custom model numpy adapter."""
+
+from openfl.plugins.frameworks_adapters.framework_adapter_interface import (
+    FrameworkAdapterPluginInterface,
+)
+
+
+class CustomFrameworkAdapter(FrameworkAdapterPluginInterface):
+    """Framework adapter plugin class."""
```

Comment:
Is this the subclassing needed for JAX or numpy?

Reply:
`get_tensor_dict` and `set_tensor_dict` method is overridden in this custom adapter class to load and set weights. 
Required at the time when collaborator returns weights for aggregation.
Although the adapters are defined in `openfl.plugins.frameworks_adapters`, custom adapters are defined in `interactive_api` examples. Same is the case for other examples as well.
Need to streamline this?

Reply:
My primary concern is, that this subclassing should work for all JAX models and not just linear regression.

Since you're adding a JAX example, your framework adapter should be able to handle other model architectures as well. For instance, a `pytorch` `model.state_dict()` can be iteratively mapped for each layer regardless of the architecture.

I'll let other reviewers chime in if this needs to be enforced or if they would like to archive it for future discussions. If latter, I'm ok and this can be closed.